### DISCUSSION TITLE
Material 3 - First tests

### DISCUSSION THREAD

**Discussion by Jean-BaptisteC** [organicmaps/organicmaps#4676] [Ideas]
Somes issues are opened to fix and improve theming and dark mode in Android. But it's a lot of work, because we need to migrate elements and be sure theming works properly on all devices.

I have doing tests to know, if migrate to Material 3 theme need a lot of refactoring and corrections.
|||
|---|---|
| <img src="https://user-images.githubusercontent.com/87148630/222534448-fcbd75e4-a806-4488-83d8-6a35b85414b3.png" height=500 /> | <img src="https://user-images.githubusercontent.com/87148630/222535309-43be0784-bd63-4edd-ae82-66173412f3c1.png" height=500 /> | |
|<img src="https://user-images.githubusercontent.com/87148630/222535861-baa2f993-bd4c-4f0c-9945-3d2c89a4d841.png" height=500 /> | <img src="https://user-images.githubusercontent.com/87148630/222536134-4d6bd98c-8435-442c-b96c-c33a9b3b714c.png" height=500 /> |
| <img src="https://user-images.githubusercontent.com/87148630/222536284-f1f98997-2311-4290-8bf1-e9180a544da2.png" height=500 /> | <img src="https://user-images.githubusercontent.com/87148630/222536405-c176e694-9edc-4622-80de-8c536d1f7657.png" height=500 />

To test, i have update theme parent in style https://github.com/Jean-BaptisteC/organicmaps/tree/m3
- Place page works properly with material 3, thanks @arnaudvergnet 
- #4444 is the first step to improve theming
- I don't know, if we need to refactor all styles (keep light and dark theme or using daynight theme)
- I think we need to refactor some things on code (Edittext, settings to change theme, implement oled mode, fix dark mode on alert dialog) before to migrate to material 3

**Comment by arnaudvergnet** [comment]
The current theme is all over the place, I cleaned up unused keys a while back but we could benefit from some more cleaning/restructuring. I think the material 3 migration is a good moment to do that!

IMO it would be better from the user perspective to make the whole material 3 migration in a single update. This way the app's design stays consistent.

But I don't think we are ready to make such a migration yet with the current state of the code.

To make the migration easier, we first need to focus on fixing bugs and refactoring the theme. It would also help to use as much as possible standard libraries/components which play nicely with material style and avoid strange custom implementations.

**Comment by biodranik** [comment]
Black layers, zoom buttons, and bottom icons in the light theme don't look good. Why did the color change?

**Reply by Jean-BaptisteC** [reply]
I didn't have update all colors in themes and default material 3 used violet like ColorAccent

**Comment by Jean-BaptisteC** [comment]
I have doing new tests today
|Light|Dark|
|-|-|
|<img src="https://github.com/organicmaps/organicmaps/assets/87148630/b3fb7b0f-2c56-493c-9fec-58275493253b" height=400 />|<img src="https://github.com/organicmaps/organicmaps/assets/87148630/315b9af4-2fb2-426b-bed6-cfd843a8a272" height=400 />|
|<img src="https://github.com/organicmaps/organicmaps/assets/87148630/403e0ae9-e192-406a-b598-9c5f86b2ec45" height=400 />|<img src="https://github.com/organicmaps/organicmaps/assets/87148630/c035cc7d-a4dd-4ea9-8468-7f6633ea1cda" height=400 />|

**Reply by biodranik** [reply]
It would be great to see old screens too.
